File: VB/LinqBinding/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace LinqBinding
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private context As New NorthwindDataContext()

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim products As New BindingSource(context.Products, "")
			Dim categories As New BindingSource(context.Categories, "")
			repositoryItemLookUpEdit1.DataSource = categories
			gridControl1.DataSource = products
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			context.SubmitChanges()
		End Sub
	End Class
End Namespace

File: CS/LinqBinding/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace LinqBinding {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        NorthwindDataContext context = new NorthwindDataContext();

        private void Form1_Load(object sender, EventArgs e) {
            BindingSource products = new BindingSource(context.Products, "");
            BindingSource categories = new BindingSource(context.Categories, "");
            repositoryItemLookUpEdit1.DataSource = categories;
            gridControl1.DataSource = products;
        }

        private void simpleButton1_Click(object sender, EventArgs e) {
            context.SubmitChanges();
        }
    }
}
